### PR TITLE
chore(auto-join): verify org founder or billing leads

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/8988

### To Test

- [ ] Sign-in with your Parabol gmail account and create a team call `AutoJoin team`
- [ ] Run `updateAutoJoin` to add the autoJoin flag to the AutoJoin team
- [ ] Run `addApprovedOrganizationDomains` for the `parabol.co` domain
- [ ] Sign-up with your email + something after it, e.g. `nick+autojoin@parabol.co`
- [ ] Verify your email address and see that you automatically join the `AutoJoin team`

### CODE REVIEW

**Comment by nickoferrall** [comment]
Thanks @Dschoordsch, I've updated the testing steps 👍

File: packages/server/dataloader/customLoaderMakers.ts [hunk 1]
```diff
@@ -698,14 +700,79 @@ export const samlByOrgId = (parent: RootDataLoader) => {
...
+        validOrgs.find((validOrg) => validOrg.id === orgId) ? [orgId] : []
+      )
+    },
+    {
+      ...parent.dataLoaderOptions
+    }
+  )
+}
+
+export const autoJoinTeamsByValidOrgId = (parent: RootDataLoader) => {
```
Comment (Dschoordsch) [inline packages/server/dataloader/customLoaderMakers.ts:764]: +1 I think it would be fair to still call it `autoJoinTeamsByOrgId`

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
I did some [tests](https://github.com/ParabolInc/parabol/pull/9036) and it's failing for an inactive founder.

**Comment by nickoferrall** [comment]
Thanks for adding the tests! The problem was I was getting the `founder` using: `org('members').filter({inactive: false}).nth(0)`, but I wanted to get the first user and could then check if that user is inactive.

**Review by Dschoordsch — APPROVED** [review]

**Pull request merged**